File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        max_v = 0
        left = 0
        right = len(heights) -  1

        while(left < right):
            max_v = max(max_v, min(heights[left], heights[right]) * (right - left))
            if heights[left] < heights[right]:
                left += 1
            else:
                right -= 1
        
        return max_v

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int len = s.length();
        int left = 0;
        int right = 0;
        int ans = 0;
        unordered_set<char> exist;

        while(left <= right && right < len){
            while(left < right && exist.find(s[right]) != exist.end()){
                exist.erase(s[left]);
                left++;
            }
            exist.insert(s[right]);
            right++;
            ans = max(ans, right-left);
        }

        return ans;

    }
};
